### PR TITLE
Add Krush65 Hotswap

### PR DESCRIPTION
**Pull request by MaiTheSan**
Add support for Krush65 Hotswap PCB

## Description

Add sawnsprojects/krush/krush65/hotswap keyboard

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [X] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by drashna** [comment]
Could you move the encoder action stuff to the via keymap, only?

**Comment by MaiTheSan** [comment]
> Could you move the encoder action stuff to the via keymap, only?

ofc :>

File: keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE  = yes       # Console for debug
+COMMAND_ENABLE  = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes
+
+# Added encoder Action
+SRC +=  encoder_actions.c
```
Comment (drashna) [inline keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk:21]:
```suggestion
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/keymaps/via/rules.mk [hunk 2]
```diff
@@ -0,0 +1,2 @@
+VIA_ENABLE = yes
+LTO_ENABLE = yes
```
Comment (drashna) [inline keyboards/sawnsprojects/krush/krush65/hotswap/keymaps/via/rules.mk:2]:
```suggestion
LTO_ENABLE = yes

```suggestion
# Added encoder Action
SRC +=  encoder_actions.c
```
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/hotswap.c [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+void matrix_scan_kb(void) {
+    encoder_action_unregister();
+    matrix_scan_user();
+}
+
+bool encoder_update_kb(uint8_t index, bool clockwise) {
+//    if (!encoder_update_user(index, clockwise)) return false;
+    encoder_action_register(index, clockwise);
+    return true;
+};
```
Comment (drashna) [inline keyboards/sawnsprojects/krush/krush65/hotswap/hotswap.c:28]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) {
        if (clockwise) {
            tap_code_delay(KC_VOLU, 10);
        } else {
            tap_code_delay(KC_VOLD, 10);
        }
    }
    return true;
}
#endif
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/keymaps/via/keymap.c [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  
+    ),
+    [3] = LAYOUT_via_rotary(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, 
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS  
+    ),
+};
```
Comment (drashna) [inline keyboards/sawnsprojects/krush/krush65/hotswap/keymaps/via/keymap.c:51]:
```suggestion
};

void matrix_scan_user(void) {
    encoder_action_unregister();
}

bool encoder_update_user(uint8_t index, bool clockwise) {
    encoder_action_register(index, clockwise);
    return false; // override kb default behavior
};
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE  = yes       # Console for debug
+COMMAND_ENABLE  = no        # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk:15]:
```suggestion
CONSOLE_ENABLE = yes        # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
NKRO_ENABLE = no            # Enable N-Key Rollover
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/readme.md [hunk 6]
```diff
@@ -0,0 +1,15 @@
+# Krush65 Hotswap
+
+![Krush65](https://imgur.com/56ZvWUm.jpg)
```
Comment (fauxpark) [inline keyboards/sawnsprojects/krush/krush65/hotswap/readme.md:3]:
```suggestion
![Krush65](https://imgur.com/56ZvWUmh.jpg)
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/krush/krush65/hotswap/readme.md [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+
+![Krush65](https://imgur.com/56ZvWUmh.jpg)
+
+A 65% keyboard PCB made and sold by SawnsProjects with Nuxros in [mechkey.store](https://mechkey.store/blogs/store-updates/introducing-krush-65). 
+
+* Keyboard Maintainer: [Mai The San](https://github.com/MaiTheSan)
+* Hardware Supported: Krush65
+* Hardware Availability: [Facebook](https://mechkey.store/products/krush-65-hotswap-pcb)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/sawnsprojects/krush/krush65/hotswap/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

File: keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = yes        # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # Enable N-Key Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = yes       # Enable keyboard RGB underglow
+ENCODER_ENABLE = yes
```
Comment (fauxpark) [inline keyboards/sawnsprojects/krush/krush65/hotswap/rules.mk:18]:
```suggestion
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/readme.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
+
+Make example for this keyboard (after setting up your build environment):
+
+    make sawnsprojects/krush65/Hotswap:default
```
Comment (fauxpark) [inline keyboards/sawnsprojects/krush/krush65/hotswap/readme.md:21]:
```suggestion
    make sawnsprojects/krush65/hotswap:default
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

File: keyboards/sawnsprojects/krush/krush65/hotswap/info.json [hunk 10]
```diff
@@ -0,0 +1,157 @@
...
+                { "label": "", "x": 6, "y": 0.75 },
+                { "label": "", "x": 7, "y": 0.75 },
+                { "label": "", "x": 8, "y": 0.75 },
+                { "label": "", "x": 9, "y": 0.75 },
+                { "label": "", "x": 10, "y": 0.75 },
+                { "label": "", "x": 11, "y": 0.75 },
+                { "label": "", "x": 12, "y": 0.75 },
+                { "label": "", "x": 13, "y": 0.75 },
+                { "label": "", "x": 14, "y": 0.75 },
+                { "label": "", "x": 15, "y": 0.75 },
```
Comment (noroadsleft) [inline keyboards/sawnsprojects/krush/krush65/hotswap/info.json:99]:
```suggestion
                { "label": "", "x": 0, "y": 0.75 },
                { "label": "", "x": 1, "y": 0.75 },
                { "label": "", "x": 2, "y": 0.75 },
                { "label": "", "x": 3, "y": 0.75 },
                { "label": "", "x": 4, "y": 0.75 },
                { "label": "", "x": 5, "y": 0.75 },
                { "label": "", "x": 6, "y": 0.75 },
                { "label": "", "x": 7, "y": 0.75 },
                { "label": "", "x": 8, "y": 0.75 },
                { "label": "", "x": 9, "y": 0.75 },
                { "label": "", "x": 10, "y": 0.75 },
                { "label": "", "x": 11, "y": 0.75 },
                { "label": "", "x": 12, "y": 0.75 },
                { "label": "", "x": 13, "y": 0.75 },
                { "label": "", "x": 14, "y": 0.75 },
                { "label": "", "x": 15, "y": 0.75 },
                { "label": "", "x": 14.75, "y": 0, "w": 0.75, "h": 0.75 },
                { "label": "", "x": 15.5, "y": 0, "w": 0.75, "h": 0.75 },
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/info.json [hunk 11]
```diff
@@ -0,0 +1,157 @@
+{
+    "keyboard_name": "Krush65 Hotswap",
+    "url": "",
+    "maintainer": "Mai The San",
```
Comment (noroadsleft) [inline keyboards/sawnsprojects/krush/krush65/hotswap/info.json:4]: This field is meant to reference your GitHub username.

```suggestion
    "maintainer": "MaiTheSan",
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/readme.md [hunk 12]
```diff
@@ -0,0 +1,23 @@
+# Krush65 Hotswap
+
+![Krush65](https://imgur.com/56ZvWUmh.jpg)
+
+A 65% keyboard PCB made and sold by SawnsProjects with Nuxros in [mechkey.store](https://mechkey.store/blogs/store-updates/introducing-krush-65). 
+
+* Keyboard Maintainer: [Mai The San](https://github.com/MaiTheSan)
+* Hardware Supported: Krush65
+* Hardware Availability: [Facebook](https://mechkey.store/products/krush-65-hotswap-pcb)
```
Comment (noroadsleft) [inline keyboards/sawnsprojects/krush/krush65/hotswap/readme.md:9]:
```suggestion
* Hardware Availability: [mechkey.store](https://mechkey.store/products/krush-65-hotswap-pcb)
```

File: keyboards/sawnsprojects/krush/krush65/hotswap/readme.md [hunk 13]
```diff
@@ -0,0 +1,23 @@
...
+
+Enter the bootloader in 3 ways:
+
+* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
+* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
+* **Keycode in layout**: Press the key mapped to `RESET` if it is available
+
+Make example for this keyboard (after setting up your build environment):
+
+    make sawnsprojects/krush65/hotswap:default
```
Comment (noroadsleft) [inline keyboards/sawnsprojects/krush/krush65/hotswap/readme.md:21]:
```suggestion
    make sawnsprojects/krush/krush65/hotswap:default
```

**Review by noroadsleft — CHANGES_REQUESTED** [review]

**Review by noroadsleft — APPROVED** [review]

**Comment by noroadsleft** [comment]
Thanks!

**Pull request merged**